scipy/stats/_distn_infrastructure.py

```diff
@@ -2652,7 +2663,7 @@ def fit(self, data, *args, **kwds):
             raise TypeError("Too many input arguments.")
 
         if not np.isfinite(data).all():
-            raise RuntimeError("The data contains non-finite values.")
+            raise ValueError("The data contains non-finite values.")
```

Comment:
@ev-br @tupui I thought I should correct this other `RuntimeError` when the data contains non-finite values. This is definitely a `ValueError`.
This breaks some tests, so before going through the trouble of changing them, I thought I'd ask your opinion on this. I think `RuntimeError` was "wrong" before, so I think it's ok to "correct" it to `ValueError` without a deprecation cycle. But is it worth the churn?

There was a discussion about what should happen here in gh-10300. They decided to raise an error to be consistent with `nan_policy='raise'`. To truly be consistent with `nan_policy='raise'`, this should be a `ValueError`.

Reply:
Yes `RuntimeError` is not consistent with what we usually do so I would change it. True that some user might do the right thing in their code and catch the exception. But I think it's fine since we don't declare any API on raise types. (Maybe we should, NumPy does it).

Reply:
Updated the tests. See gh-10300 and associated PR gh-10480 for some background. @rlucas7 @chrisb83 do you agree that this should be a ValueError? It wasn't documented, so I think we should change it. Doing that now would smooth the way toward adding `nan_policy` for `fit` someday.

Reply:
I agree

Reply:
Thanks @chrisb83 @tupui!